internal/atomiccounter/atomiccounter.go

```diff
@@ -0,0 +1,28 @@
+package atomiccounter
+
+import (
+	"sync/atomic"
+)
+
+type AtomicCounter struct {
+	val int64
+}
+
+// NewAtomicCounter returns a new counter with the default value of 0.
+func NewAtomicCounter() AtomicCounter {
+	return AtomicCounter{}
+}
+
+func (ac *AtomicCounter) Increment() {
+	atomic.AddInt64(&ac.val, 1)
+}
+
+func (ac *AtomicCounter) Decrement() {
+	atomic.AddInt64(&ac.val, -1)
+}
+
+// Get is not safe to use for synchronizing work between goroutines.
+// It is just for logging the current value.
```

Comment:
Do we want to add a mutex lock on this to ensure safety? I'm honestly not super concerned

Reply:
If we are agreeing on adding a `mutex`, I would consider use channel for better performance. I have talked with Adam about this but he would give a better explanation than me.

Reply:
I don't think it's necessary to add a locking mechanism for this. I'm indifferent

Reply:
No, IMO atomic inc/dec is good enough.
Just wanted to clear that just because this _thing_ works for this specific use case, doesn't mean it should be used for synchronization.

There is no behavior/decision based on the value of the counter, just logging it to file so even if by the time a user sees the value in the file the actual count has changed it isn't a big deal.

Reply:
@adam-mateen agreed

Reply:
Specifically don't use it like this:
```
var lock = atomiccounter.NewAtomicCounter()

// foo tries to do some EXCLUSIVE work every second
func foo() {
    for {
        if lock.Get() == 0 {
            lock.Increment()
            // do work
            .
            .
            .
        }
        time.Sleep(time.Second)
    }
}
```
if 2 or more goroutines call `foo()`, _you're going to have a bad time_.

get() plus increment() is not the same as `atomic.CompareAndSwapInt64`.
And if that is what you want, you should just use a mutex!